pkg/controller/config/config_controller.go

```diff
@@ -839,11 +839,7 @@ func sourceBasedOnRecursion(path string) mf.Source {
 }
 
 func createAddonsPipelines(mgr manager.Manager) (mf.Manifest, error) {
-	pipelineTemplate, err := mf.NewManifest("deploy/resources/templates/pipelinetemplate.yaml")
-	if err != nil {
-		return mf.Manifest{}, err
-	}
-	pipelines, err := addons.CreatePipelines(pipelineTemplate, mgr.GetClient())
+	pipelines, err := addons.CreatePipelines(flag.TemplatePath, mgr.GetClient())
```

Comment:
if we are passing `flag.TemplatePath` here, then shall we use `flag.TemplatePath` directly inside the function.
Is it necessary to to pass it from the controller.

Reply:
in fact the `createAddonsPipelines` is unnecessary in this update. 
In `readAddons` (in config_controller.go), we can directly call `addons.CreatePipelines`. Then read `falg.TemplatePath` from addons package.

Reply:
It was intentional to pass the template path, as I was facing issues while adding tests for it. I am not sure why the test was not able to read the resource in `deploy/resources/templates`.
Please let me know if there is something known.

Reply:
Oh, you are right, we update the function. will remove it and call directly.